ConsistencyManager/DataStructures/WeakModelUpdatesListenerArray.swift

```diff
@@ -0,0 +1,170 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import Foundation
+
+
+/**
+ This is a temporary workaround to an Apple bug. We can't specify a WeakArray<Protocol>, so I have to copy-paste it for a specific type.
```

Comment:
Is there something that prevents us from doing `Array<WeakHolder<Protocol>>`

If that doesn't work then `Array<WeakHolderForProtocolA>`

Implementing WeakHolderForProtocolA, WeakHolderForProtocolB, ... etc would be much simpler than reimplementing the entire array API for each protocol.